changes.go

```diff
@@ -31,13 +31,18 @@ type ArchConstraint interface {
 // ConstraintGetter represents a architecture constraint parser.
 type ConstraintGetter func(string) ArchConstraint
 
+// RevisionGetter resolves the revision of a charm from the list of parameters.
+type RevisionGetter func(string, string, string, string) (int, error)
```

Comment:
Please name the args so there is no mix-up with their order as all of them are strings.